hummingbot/strategy/avellaneda_market_making/avellaneda_market_making_config_map_pydantic.py

```diff
@@ -0,0 +1,529 @@
+        description="Allows activating multi-order levels.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: f"Select the order levels mode ({'/'.join(OrderLevelsMode)}",
+        ),
+    )
+    order_override: Optional[Dict] = Field(
+        default=None,
+        description="Allows custom specification of the order levels and their spreads and amounts.",
+        client_data=None,
+    )
+    hanging_orders_mode: Union[TrackHangingOrdersModel, IgnoreHangingOrdersModel] = Field(
+        default=IgnoreHangingOrdersModel.construct(),
+        description="When tracking hanging orders, the orders on the side opposite to the filled orders remain active.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: f"How do you want to handle hanging orders? ({'/'.join(HangingOrdersMode)})",
+        ),
+    )
+    should_wait_order_cancel_confirmation: bool = Field(
+        default=True,
+        description=(
+            "If activated, the strategy will await cancellation confirmation from the exchange"
+            " before placing a new order."
+        ),
+        client_data=ClientFieldData(
+            prompt=lambda mi: (
+                "Should the strategy wait to receive a confirmation for orders cancellation"
+                " before creating a new set of orders?"
+                " (Not waiting requires enough available balance) (Yes/No)"
+            ),
+        )
+    )
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        required_exchanges.append(self.exchange)
+
+    class Config:
+        validate_assignment = True
+
+    @validator("exchange", pre=True)
+    def validate_exchange(cls, v: str):
+        ret = validate_exchange(v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+    @validator("market", pre=True)
+    def validate_exchange_trading_pair(cls, v: str, values: Dict):
+        exchange = values.get("exchange")
+        ret = validate_market_trading_pair(exchange, v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+    @validator("execution_timeframe_model", pre=True)
+    def validate_execution_timeframe(
+        cls, v: Union[str, InfiniteModel, FromDateToDateModel, DailyBetweenTimesModel]
+    ):
+        if not isinstance(v, Dict) and not hasattr(ExecutionTimeframe, v):
+            raise ValueError(
+                f"Invalid timeframe, please choose value from {[e.value for e in list(ExecutionTimeframe)]}"
+            )
+        elif isinstance(v, (InfiniteModel, FromDateToDateModel, DailyBetweenTimesModel)):
+            sub_model = v
+        elif v == ExecutionTimeframe.infinite:
+            sub_model = InfiniteModel.construct()
+        elif v == ExecutionTimeframe.daily_between_times:
+            sub_model = DailyBetweenTimesModel.construct()
+        elif v == ExecutionTimeframe.from_date_to_date:
+            sub_model = FromDateToDateModel.construct()
```

Comment:
If we have a dictionary mapping the class title with the class (can be generated easily if the class stores the value as a class variable, as suggested before) we can eliminate all this `if` (or replace them with only one that checks the string is in the dictionary's keys).

Reply:
Addressed